inc/cloudfront_media_purge/namespace.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Altis\Cloud\Cloudfront_Media_Purge;
+
+use Exception;
+use function Altis\get_aws_sdk;
+
+function bootstrap() {
+	add_action( 'delete_attachment', __NAMESPACE__ . '\\purge_media_file_cache', 100 );
+}
+
+/**
+ * Return an AWS CloudFront Client instance
+ *
+ * @return \Aws\CloudFront\CloudFrontClient
+ */
+function get_aws_client() {
+	return get_aws_sdk()->createCloudFront( [
+		'version' => 'latest',
+	] );
+}
+
+/**
+ * Purge an attachment file cache from CloudFront
+ *
+ * It sends a request to invalidate all sizes given for a media file
+ *
+ * @param int $media_id Attachment ID.
+ *
+ * @return bool
+ */
+function purge_media_file_cache( $media_id ) {
+	$meta = wp_get_attachment_metadata( $media_id );
+	if ( ! $meta ) {
+		return false;
+	}
+	$baseurl = '/tachyon/';
+	$basedir = trailingslashit( dirname( $meta['file'] ) );
+	$items   = [ $baseurl . $meta['file'] . '*' ];
+
+	if ( ! empty( $meta['sizes'] ) ) {
+		foreach ( $meta['sizes'] as $size => $size_meta ) {
+			$items[] = $baseurl . $basedir . $size_meta['file'] . '*';
+		}
+	}
+
+	$client = get_aws_client();
+
+	$distribution_id = apply_filters( 'altis_cloudfront_media_purge_distribution_id', defined( 'CLOUDFRONT_DISTRIBUTION_ID' ) ? CLOUDFRONT_DISTRIBUTION_ID : '' );
```

Comment:
Is this being set by ECS / EC2 automatically currently? Are we already giving all the permissions to the instance role to be able to create the invalidation?

Reply:
Nope - For ECS: I have raised PR to add  `CLOUDFRONT_DISTRIBUTION_ID`.

I added a filter in case we want to change distribution id for the domain - like the plugin use `CLOUDFRONT_MERCATOR_DISTRIBUTION_ID`